GitCommands/DiffMergeTools/DiffMergeToolConfigurationManager.cs

```diff
@@ -0,0 +1,237 @@
+        /// <returns>The command for the diff/merge tool configured in the effective config. </returns>
+        public string GetToolCommand(string toolName, DiffMergeToolType toolType)
+        {
+            if (string.IsNullOrWhiteSpace(toolName))
+            {
+                return string.Empty;
+            }
+
+            var command = GetToolSetting(toolName, toolType, "cmd");
+            if (!string.IsNullOrWhiteSpace(command))
+            {
+                return command;
+            }
+
+            var config = LoadDiffMergeToolConfig(toolName, null);
+
+            switch (toolType)
+            {
+                case DiffMergeToolType.Diff: return config.FullDiffCommand;
+                case DiffMergeToolType.Merge: return config.FullMergeCommand;
+                default: throw new NotSupportedException();
+            }
+        }
+
+        /// <summary>
+        /// Gets the path to the diff/merge tool configured in the effective config.
+        /// </summary>
+        /// <param name="toolName">The name of the diff/merge tool.</param>
+        /// <param name="toolType">Type of the tool.</param>
+        /// <returns>The path to the diff/merge tool configured in the effective config. </returns>
+        public string GetToolPath(string toolName, DiffMergeToolType toolType)
+        {
+            if (string.IsNullOrWhiteSpace(toolName))
+            {
+                return string.Empty;
+            }
+
+            var path = GetToolSetting(toolName, toolType, "path");
+            if (!string.IsNullOrWhiteSpace(path))
+            {
+                return path;
+            }
+
+            return LoadDiffMergeToolConfig(toolName, null).Path;
+        }
+
+        public DiffMergeToolConfiguration LoadDiffMergeToolConfig(string toolName, string userSuppliedPath)
+        {
+            if (string.IsNullOrWhiteSpace(toolName))
+            {
+                throw new ArgumentException(@"Invalid diff/merge tool requested", nameof(toolName));
+            }
+
+            string fullPath;
+
+            var diffTool = RegisteredDiffMergeTools.Get(toolName);
+            if (diffTool == null)
+            {
+                var exeName = toolName + ".exe";
+                if (!string.IsNullOrWhiteSpace(userSuppliedPath))
+                {
+                    fullPath = userSuppliedPath;
+                }
+                else
+                {
+                    PathUtil.TryFindFullPath(exeName, out fullPath);
+                }
+
+                return new DiffMergeToolConfiguration(exeName, fullPath ?? string.Empty, null, null);
+            }
+
+            if (!string.IsNullOrWhiteSpace(userSuppliedPath))
+            {
+                fullPath = userSuppliedPath;
+            }
+            else
+            {
+                var pathsToSearch = new[] { UnquoteString(GetToolSetting(diffTool.Name, DiffMergeToolType.Merge, "path")) }.Union(diffTool.SearchPaths);
```

Comment:
This will get the path to the mergetool with the suggested name, which may be different from the difftool
Normally this will work (as mergetool is searched also for difftool)

Reply:
What are the changes of difftool executable be different from a mergetool executable?
From my observations all tools I've seen use the same executable.